api/osf_groups/views.py

```diff
@@ -0,0 +1,231 @@
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    serializer_class = GroupSerializer
+    view_category = 'groups'
+    view_name = 'group-list'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        user = self.request.user
+        if user.is_anonymous:
+            return OSFGroup.objects.none()
+        return user.osf_groups
+
+    # overrides ListCreateAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        """Create an OSFGroup.
+
+        :param serializer:
+        """
+        # On creation, logged in user is the creator
+        user = self.request.user
+        serializer.save(creator=user)
+
+
+class GroupDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsGroupManager,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    serializer_class = GroupDetailSerializer
+    view_category = 'groups'
+    view_name = 'group-detail'
+    ordering = ('-modified', )
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def get_object(self):
+        return self.get_osf_group()
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def perform_destroy(self, instance):
+        auth = get_user_auth(self.request)
+        instance.remove_group(auth=auth)
+
+
+class OSFGroupMemberBaseView(JSONAPIBaseView, OSFGroupMixin):
+    """
+    Base group used for OSFGroupMemberList and OSFGroupMemberDetail
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsGroupManager,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+
+    model_class = apps.get_model('osf.OSFUser')
+    serializer_class = GroupMemberSerializer
+    view_category = 'groups'
+    ordering = ('-modified', )
+
+    def _assert_member_belongs_to_group(self, user):
+        group = self.get_osf_group()
+        # Checking group membership instead of permissions, so unregistered members are
+        # recognized as group members
+        if not group.is_member(user):
+            raise NotFound('{} cannot be found in this OSFGroup'.format(user._id))
+
+    def get_serializer_class(self):
+        if self.request.method == 'PUT' or self.request.method == 'PATCH' or self.request.method == 'DELETE':
```

Comment:
might just want to make `method = self.request.method` to save some space or `self.request.method  in ('PUT', 'PATCH', 'DELETE')`